index.js

```diff
@@ -1,79 +1,171 @@
+// TODO setState and getState
 const TAX = 0.08;
 const state = {
+  stepNumber: 0,
   customerBankAccount: 0,
   customerSpendLimit: 0,
-  phoneName: '',
-  phonePrice: '',
-  accessoryName: '',
-  accessoryPrice: '',
-  bundleSize: '',
+  phoneName: null,
+  phonePrice: 0,
+  accessoryName: null,
+  accessoryPrice: 0,
+  bundleSize: 0,
   totalSpent: 0,
 };
 
 initialize();
 
 function initialize() {
-  doneWithStepEl(1).onclick = () => { goToStep(2); };
-  doneWithStepEl(2).onclick = () => { goToStep(3); };
-  doneWithStepEl(3).onclick = () => { goToStep(4); };
+  goNextStep();
 
-  document.getElementById('back-step3').onclick = () => { goToPreviousStep(3); };
-  document.getElementById('back-step4').onclick = () => { goToPreviousStep(4); };
+  let count = 1;
+  while (count <= 4) {
+    nextButtonEl(count).onclick = () => { goNextStep(); };
+    count += 1;
+  }
 
-  goToStep(1);
+  backButtonEl(3).onclick = () => { goToPreviousStep(); };
+  backButtonEl(4).onclick = () => { goToPreviousStep(); };
 }
 
-function goToStep(stepNum) {
-  if (stepNum === 1 || stepNum === 2) {
-    toggleDisplaySteps(stepNum, (stepNum - 1));
-  } else if (stepNum === 3) {
-    // TODO - set validation for 0 or NaN
-    if (state.customerBankAccount === '' || state.customerSpendLimit === '') {
-      alert('Please fill out both inputs');
+
+function goNextStep() {
+  if (state.stepNumber <= 1) {
+    toggleDisplaySteps();
+  } else {
+    validateStep() && toggleDisplaySteps();
+  }
+}
+
+function validateStep() {
+  if (state.stepNumber === 2) {
+    if (state.customerBankAccount === 0 || Number.isNaN(state.customerBankAccount)
+     || state.customerSpendLimit === 0 || Number.isNaN(state.customerSpendLimit)) {
+      checkStepTwoErrors();
     } else {
+      removeStepTwoErrors();
       document.getElementById('top-bar').classList.add('active');
-      toggleDisplaySteps(stepNum, (stepNum - 1));
+
+      return true;
     }
-  } else if (stepNum === 4) {
-    if (state.phonePrice === '' || state.accessoryPrice === '' || state.bundleSize === '') {
-      // TODO - if '' or NaN (has issue if changed back to default ')')
-      alert('Please select one of each');
+  }
+  if (state.stepNumber === 3) {
+    if (state.phonePrice === 0 || state.accessoryPrice === 0 || state.bundleSize === 0) {
+      checkStepThreeErrors();
     } else {
+      removeStepThreeErrors();
       overUnderForSummary();
-      toggleDisplaySteps(stepNum, (stepNum - 1));
+
+      return true;
     }
   }
 }
 
-function doneWithStepEl(stepNum) {
-  return document.getElementById(`done-step${stepNum}`);
+function checkStepTwoErrors() {
+  if (state.customerBankAccount === 0 || Number.isNaN(state.customerBankAccount)) {
+    document.getElementById('bank-account').classList.add('error');
+    document.getElementById('bank-error').classList.add('active');
```

Comment:
There's a pattern of adding a class to the input and then a class to the error, making them essentially independent without any centralized state. This won't result in errors currently, but it could easily get out of sync through future changes.

Ideally you want to indicate error state in one place. In your HTML, this would probably happen by putting each input and it's error message into a dedicated wrapper element, eg. a `div`, and then you just add/remove an `error` class on that one element. The input and error message within that wrapper would then just key off of that class on the parent via CSS.

Reply:
Thanks, that helped cut the error section in half...yikes